{vis}[foss/2023a] napari v0.4.19.post1, VisPy v0.14.1  w/ Python v3.11.3

ThomasHoffmann77 opened the pull request:
(created using `eb --new-pr`)

ThomasHoffmann77 commented:
> see also: #19652 Assuming you're also working on RELION 5, I have more PRs coming

see #19240 - did not port it to 2023a, yet. looking forward to your PRs.

regarding napari 0.4.18 vs 0.4.19rc3. The pydantic < 2 requirement might lead to conflicts. 4.19 uses pydantic_compat.

jfgrimm commented:
> > see also: #19652 Assuming you're also working on RELION 5, I have more PRs coming
> 
> see #19240 - did not port it to 2023a, yet. looking forward to your PRs.
> 
> regarding napari 0.4.18 vs 0.4.19rc3. The pydantic < 2 requirement might lead to conflicts. 4.19 uses pydantic_compat.

made a draft here: https://github.com/easybuilders/easybuild-easyconfigs/pull/19678

jfgrimm commented:
@ThomasHoffmann77 can you fix the conflicts please?

ThomasHoffmann77 commented:
> @ThomasHoffmann77 can you fix the conflicts please?
@jfgrimm I deleted PyOpenGL, Qtconsole, and QtPy. I updated some extensions and to 0.4.19.post1.
I opened a new PR #20007 to add PyOpenGL-accelerate.

ThomasHoffmann77 commented:
@jfgrimm I downgraded the VisPy dependency from 0.14.1 to 0.12.2 since there was a conflict with the napari 0.4.18 easyconfig. Preferably I'd like to keep the newer version of VisPy.

ThomasHoffmann77 commented:
Test report by @thomashoffmann77
**FAILED**
Build succeeded for 1 out of 3 (6 easyconfigs in total)
srv-mahamid-01.embl.de - Linux AlmaLinux 8.8, x86_64, AMD EPYC 7513 32-Core Processor, 2 x NVIDIA NVIDIA GeForce RTX 3090, 535.113.01, Python 3.6.8
See https://gist.github.com/ThomasHoffmann77/dfcccd87696ced64a8cad347055edad6 for a full test report.

ThomasHoffmann77 commented:
Test report by @thomashoffmann77
**FAILED**
Build succeeded for 0 out of 2 (6 easyconfigs in total)
srv-mahamid-01.embl.de - Linux AlmaLinux 8.8, x86_64, AMD EPYC 7513 32-Core Processor, 2 x NVIDIA NVIDIA GeForce RTX 3090, 535.113.01, Python 3.6.8
See https://gist.github.com/ThomasHoffmann77/9f8d87068c021bbbee47eeb18df266e0 for a full test report.

ThomasHoffmann77 commented:
Test report by @thomashoffmann77
**FAILED**
Build succeeded for 0 out of 2 (7 easyconfigs in total)
srv-mahamid-01.embl.de - Linux AlmaLinux 8.8, x86_64, AMD EPYC 7513 32-Core Processor, 2 x NVIDIA NVIDIA GeForce RTX 3090, 535.113.01, Python 3.6.8
See https://gist.github.com/ThomasHoffmann77/1ef026d81ffb48d4265e348c2321a22a for a full test report.

ThomasHoffmann77 commented:
Test report by @thomashoffmann77
**SUCCESS**
Build succeeded for 0 out of 0 (5 easyconfigs in total)
srv-mahamid-01.embl.de - Linux AlmaLinux 8.8, x86_64, AMD EPYC 7513 32-Core Processor, 2 x NVIDIA NVIDIA GeForce RTX 3090, 535.113.01, Python 3.6.8
See https://gist.github.com/ThomasHoffmann77/caa9f3a0d67e54b4f542c7b5036625f6 for a full test report.

ThomasHoffmann77 commented:
Test report by @thomashoffmann77
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
srv-mahamid-01.embl.de - Linux AlmaLinux 8.8, x86_64, AMD EPYC 7513 32-Core Processor, 2 x NVIDIA NVIDIA GeForce RTX 3090, 535.113.01, Python 3.6.8
See https://gist.github.com/ThomasHoffmann77/b0a2112a1f875e261fc111fdcf7465a7 for a full test report.

jfgrimm commented on easybuild/easyconfigs/v/VisPy/VisPy-0.14.1-foss-2023a.eb:
```diff
@@ -0,0 +1,44 @@
+
+use_pip = True
+
+local_inject_version = """sed -i "s/use_scm_version/version='%(version)s', &/g" setup.py &&"""
+
+exts_list = [
+    ('freetype_py', '2.4.0', {
+        'preinstallopts': local_inject_version,
+        'modulename': 'freetype',
+        'sources': ['freetype-py-2.4.0.zip'],
+        'checksums': ['8ad81195d2f8f339aba61700cebfbd77defad149c51f59b75a2a5e37833ae12e'],
+    }),
+    ('hsluv', '5.0.3', {
+        'checksums': ['2586bcb61d29d76e89e563a6836df24d86939961c9657f129a59f7617de45377'],
+    }),
+    ('vispy', version, {
+        'preinstallopts': local_inject_version,
+        'checksums': ['249a50979fc00a8b65109283354dcf12cf415c1a5dcf9821e113f6e590b9b93c'],
+        'use_pip': True
+    }),
+]
```
@ThomasHoffmann77 this workaround shouldn't be needed since we moved `setuptools_scm` into the main `Python` easyconfig
```suggestion
exts_list = [
    ('freetype_py', '2.4.0', {
        'modulename': 'freetype',
        'sources': ['freetype-py-2.4.0.zip'],
        'checksums': ['8ad81195d2f8f339aba61700cebfbd77defad149c51f59b75a2a5e37833ae12e'],
    }),
    ('hsluv', '5.0.3', {
        'checksums': ['2586bcb61d29d76e89e563a6836df24d86939961c9657f129a59f7617de45377'],
    }),
    ('vispy', version, {
        'checksums': ['249a50979fc00a8b65109283354dcf12cf415c1a5dcf9821e113f6e590b9b93c'],
        'use_pip': True
    }),
]
```

jfgrimm reviewed (changes requested):
sorry for not looking at this again sooner

jfgrimm commented:
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node086.viking2.yor.alces.network - Linux Rocky Linux 8.9, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/5e5fc117102596c1ce98dd8b4db9cc7a for a full test report.

jfgrimm commented:
@boegelbot: please test @ generoso

jfgrimm commented:
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node086.viking2.yor.alces.network - Linux Rocky Linux 8.9, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/6935c9f6f279db28017d86eccdee10ce for a full test report.

jfgrimm reviewed (dismissed):
LGTM (just waiting on the bot reports + CI)

jfgrimm commented:
jsc-zen3 build failed with:
```
asyncpg 0.29.0 has requirement async-timeout>=4.0.3; python_version < "3.12.0", but you have async-timeout 4.0.2.
```

jfgrimm commented:
@boegelbot: please test @ generoso

jfgrimm commented:
I've traced the difference between my installations and the bots to the following PR: https://github.com/easybuilders/easybuild-easyconfigs/pull/20511

jfgrimm commented:
@boegelbot: please test @ generoso

jfgrimm commented:
@boegelbot: please test @ jsc-zen3

ThomasHoffmann77 commented:
@jfgrimm are the sanity pip check messages about missing hatchling related to 20546 and jupyter-server?

jfgrimm commented:
Going in, thanks @ThomasHoffmann77!
